File: solutions-blind/python/005-rle/solution.py
def encode(s: str) -> str:
    if not s:
        return ""
    out = []
    i = 0
    while i < len(s):
        ch = s[i]
        j = i + 1
        while j < len(s) and s[j] == ch:
            j += 1
        out.append(ch + str(j - i))
        i = j
    return "".join(out)


def decode(s: str) -> str:
    if s == "":
        return ""
    out = []
    i = 0
    while i < len(s):
        ch = s[i]
        i += 1
        if i >= len(s) or not s[i].isdigit():
            raise ValueError("missing count after character")
        start = i
        while i < len(s) and s[i].isdigit():
            i += 1
        n = int(s[start:i])
        if n < 1:
            raise ValueError("count must be at least 1")
        out.append(ch * n)
    return "".join(out)

File: solutions-blind/javascript/005-rle/solution.mjs
export function encode(s) {
  if (s === "") return "";
  let result = "";
  let i = 0;
  while (i < s.length) {
    const ch = s[i];
    let j = i + 1;
    while (j < s.length && s[j] === ch) j++;
    result += ch + String(j - i);
    i = j;
  }
  return result;
}

export function decode(s) {
  if (s === "") return "";
  let result = "";
  let i = 0;
  while (i < s.length) {
    const ch = s[i++];
    if (i >= s.length || s[i] < "0" || s[i] > "9") {
      throw new Error("missing count after character");
    }
    let countStr = "";
    while (i < s.length && s[i] >= "0" && s[i] <= "9") {
      countStr += s[i++];
    }
    const n = Number(countStr);
    if (n < 1) throw new Error("count must be at least 1");
    result += ch.repeat(n);
  }
  return result;
}
